src/sfctl/tests/help_text_test.py

```diff
@@ -0,0 +1,364 @@
+
+        Limitations: This test doesn't search for new commands which are added.
+                     If a test entry is not added here, then that entry will not be
+                     verified.
+                     While we can read in the lines to make this test more automated,
+                     such that no subcommands or groups need to be listed, that case
+                     will not catch whether or not a command or subgroups has been
+                     accidentally removed.
+
+        command_input (string): This represents the command for which you want to get the help text.
+            For example, "sfctl" or "sfctl application" or "sfctl application list".
+            Parameter command_input should not include the "-h" to get the help text, as this
+            method will take care of that.
+
+        subgroups (tuple of strings): This represents all of the subgroups expected in the
+            help text. This tuple must be in alphabetical order.
+
+        commands (tuple of strings): This represents all of the commands expected in the
+            help text. This tuple must be in alphabetical order.
+
+        Help text has one of the two following formats:
+
+        1.  Group
+                sfctl chaos: Start, stop and report on the chaos test service.
+
+            Subgroups:
+                chaos      : Start, stop and report on the chaos test service.
+                cluster    : Select, manage and operate Service Fabric clusters.
+
+            Commands: [This section may not exist depending on input]
+                get     : Gets something.
+                events  : Gets the next segment of the Chaos events...
+                      range.
+
+        or
+
+        2.  Command
+                sfctl application create: Creates a Service Fabric appli...
+
+            Arguments
+                --app-version [Required]: The version of the application type as defined in the
+                                          manifest.
+                --timeout -t            : Server timeout in seconds.  Default: 60.
+
+            Global Arguments
+                --debug                 : Increase logging verbosity to show all debug logs.
+        """
+
+        try:
+            # This variable tracks what sections of the help text we are in
+            # Possibilities are Group, Subgroups, Commands, Command, Arguments,
+            # and Global Arguments.
+            # Once we no longer support python 2, change section options of enums
+            section = 'Start'
+
+            # A tracker to know which subgroup or command has appeared in help text
+            subgroups_index = 0
+            commands_index = 0
+
+            help_command = command_input + ' -h'
+
+            # Call the provided command in command line
+            # Do not split the help_command, as that breaks behavior: Linux ignores the splits.
+            pipe = Popen(help_command, shell=True, stdout=PIPE, stderr=PIPE)
```

Comment:
Could we use a ScenarioTest here from `knack`? It seems like having to manually start a process is a bit of a hassle, there's already logic there to invoke commands inside the CLI environment.

Reply:
I tried this, but ran into issues because the command errored out. My guess is that it expected some return value but didn't get anything.